File: Assets/Scripts/Lua/Controller/MainFormCtr.lua
-- region *.lua
-- Date
-- 此文件由[BabeLua]插件自动生成



-- endregion


MainFormCtr = class(FormCtr)

-- 加载
function MainFormCtr:OnLoad()
    print(11);
    -- 绑定事件
    --Ap.Lua.PointerClickEvent.Create(self.m_View.btnStart, function(sender)
        --self:btnStart_OnClick(sender);
    --end );
    Ap.Lua.PointerClickEvent.Create(self.m_View.btnClose, function(sender)
        self:btnClose_OnClick(sender);
    end );
    Ap.Lua.PointerClickEvent.Create(self.m_View.btnOpen, function(sender)
        self:btnOpen_OnClick(sender);
    end );
end

-- 开始按钮点击
function MainFormCtr:btnStart_OnClick(sender)
    print("click" .. self.m_View.m_ID);
    
    -- TEST
    Ap.Base.Context.Instance.ServerIp = "127.0.0.1";
    Ap.Base.Context.Instance.ServerPort = 2345;
    
    ProtocolCtr.SendLOGIN_INFO({USER_NAME="11",USER_PWD="123",TOKEN="123"});

end

-- 关闭按钮点击
function MainFormCtr:btnClose_OnClick(sender)
    print("close" .. self.m_View.m_ID);
    FormManager.Close(self.m_View.m_ID);
end

-- 打开新的界面
function MainFormCtr:btnOpen_OnClick(sender)
    local id = FormManager.Show("MainForm", MainForm.new());
    print("open" .. id);
    FormManager.BindCtr(id, MainFormCtr.new());
end




-- 监听实体数据变更消息
EventProtocol.addEventListener("msg.COMMON_INFO",function()
    

end);

File: Assets/Scripts/CS/Controller/MainFormCtr.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Ap.Base;
using Ap.Net;
using Ap.UI;
using UnityEngine.UI;

public class MainFormCtr : MonoBehaviourEx
{
    public ButtonEx StartBtn;
    public ButtonEx LoginBtn;


    public void Start()
    {
        StartBtn.onClick.AddListener(new UnityEngine.Events.UnityAction(StartBtn_OnClick));
        Ap.Managers.NetworkManager.Instance.OnReceivedMessage += NetworkManager_OnReceivedMessage;
        LoginBtn.onClick.AddListener(new UnityEngine.Events.UnityAction(LoginBtn_OnClick));
        // test 
        ProtocolParser.AddHandle("Msg.CommonInfo", MainFormCtr_OnLogin);
    }

    private void OnDestroy()
    {
        //Ap.Managers.NetworkManager.Instance.OnReceivedMessage -= NetworkManager_OnReceivedMessage;
        ProtocolParser.RemoveHandle("Msg.CommonInfo", MainFormCtr_OnLogin);
    }



    /// <summary>
    /// 接收到网络消息
    /// </summary>
    /// <param name="id"></param>
    /// <param name="data"></param>
    private void NetworkManager_OnReceivedMessage(int id, ByteBuffer data)
    {
        if (id == Ap.Net.Protocal.Connect)
        {
            // 连接成功
            print("connect");
        }
        else if (id == Ap.Net.Protocal.Disconnect)
        {
            // 断开
        }
    }

    private void StartBtn_OnClick()
    {

        Ap.Base.Context.Instance.ServerIp = "127.0.0.1";
        Ap.Base.Context.Instance.ServerPort = 3563;
        Ap.Managers.NetworkManager.Instance.SendConnect();
    }

    private void LoginBtn_OnClick()
    {
        Msg.LoginInfo info = new Msg.LoginInfo();
        info.Token = "111";
        info.UserName = "Dada";
        info.UserPwd = "Mom";
        ProtocolParser.SendLOGIN_INFO(info);
    }

    /// <summary>
    /// 监听登陆消息
    /// </summary>
    /// <param name="data"></param>
    public static void MainFormCtr_OnLogin(object data)
    {
        if (data == null)
            return;
        var info = data as Msg.CommonInfo;

        print("LoginInfo:" + info.Msg);
        if (info.Nums != null)
        {
            foreach (var v in info.Nums)
            {
                print("LoginInfo Nums:" + v.ToString());
            }
        }
    }
}

